crates/wasmtime/src/linker.rs

```diff
@@ -31,22 +38,67 @@ use std::rc::Rc;
 /// Names in a `Linker` cannot be defined twice, but allowing duplicates by
 /// shadowing the previous definition can be controlled with the
 /// [`Linker::allow_shadowing`] method.
-pub struct Linker {
-    store: Store,
-    string2idx: HashMap<Rc<str>, usize>,
-    strings: Vec<Rc<str>>,
-    map: HashMap<ImportKey, Extern>,
+pub struct Linker<T> {
```

Comment:
It's worth pointing out that I haven't implemented the `sub_linker` API from the RFC. I'm not sure if that's actually useful to implement or whether someone wants it. The intention was to preserve the functionality where today you can create global functions via `Config` and then each store-based `Linker` can add its own set of functions after-the-fact. 

If that's used by applications and still desired I can add the `sub_linker` functionality here, but I figured it might be good to start with something simpler and only add that later if necessary.